File: Java/2024_cursoemvideo_poo/aula06_encap_controleRemoto/src/ControleRemoto.java
public class ControleRemoto implements Controlador {
    private int volume;
    private boolean ligado;
    private boolean tocando;

    public ControleRemoto() {
        this.volume = 50;
        this.ligado = false;
        this.tocando = false;
    }

    public int getVolume() {
        return volume;
    }

    public void setVolume(int volume) {
        this.volume = volume;
    }

    public boolean isLigado() {
        return ligado;
    }

    public void setLigado(boolean ligado) {
        this.ligado = ligado;
    }

    public boolean isTocando() {
        return tocando;
    }

    public void setTocando(boolean tocando) {
        this.tocando = tocando;
    }

    //Métodos Abstratos


    @Override
    public void ligar() {
        this.setLigado(true);
    }
    @Override
    public void desligar() {
        this.setLigado(false);
    }

    @Override
    public void play() {
        this.setTocando(true);
    }
    @Override
    public void pause() {
        this.setTocando(false);
    }

    @Override
    public void aumentarVolume() {
        this.setVolume(volume+1);
    }
    @Override
    public void diminuirVolume() {
        this.setVolume(volume-1);
    }

    @Override
    public void ligarMudo() {
        this.setVolume(0);
    }
    @Override
    public void desligarMudo() {
        this.setVolume(50);
    }

    @Override
    public void abrirMenu() {
        System.out.println("Ligado: " + isLigado());
        System.out.println("Volume: " + getVolume());
        System.out.println("Tocando: " + isTocando());
    }
    @Override
    public void fecharMenu() {
        System.out.println("Fechando Menu");
    }
}

File: PHP/2016_cursoemvideo_poo/aula06_encapsulamento/ControleRemoto.php
<?php 
require_once 'Controlador.php';
class ControleRemoto {
    private $volume, $ligado, $tocando;
    function __construct() {
        $this->ligado = false;
        $this->tocando = false;
        $this->volume = 50;
    }
    
    function getLigado() {return $this->ligado;}
    function setLigado($ligado) {$this->ligado = $ligado;}
    function getTocando() {return $this->tocando;}
    function setTocando($tocando) {$this->tocando = $tocando;}
    function getVolume() {return $this->volume;}
    function setVolume($volume) {$this->volume = $volume;}

    //Métodos
    public function ligar() {
        $this->setLigado(true);
    }
    public function desligar() {
        $this->setLigado(false);
    }
    public function abrirMenu() {
        echo "<p>";
        echo "Menu:<br>";
        echo "<br>Ligado: " . ($this->getLigado() ? "Sim":"Não");
        echo "<br>Tocando: " . ($this->getTocando() ? "Sim":"Não");
        echo "<br>Volume = ". $this->getVolume();
        echo "</p>";
    }
    public function fecharMenu() {
        echo " ";
    }
    public function maisVolume() {
        if ($this->getLigado() == true) {
            if ($this->getVolume() < 100) {
                $this->setVolume($this->getVolume() + 10);
            }
        }
    }
    public function menosVolume() {
        if ($this->getLigado() == true) {
            if ($this->getVolume() > 0) {
                $this->setVolume($this->getVolume() - 1);
            }
        }
    }
    public function ligarMudo() {
        if ($this->getLigado() == true) {
            if ($this->getVolume() > 0) {
                $this->setVolume(0);
            }
        }
    }
    public function play() {
        if ($this->getLigado() && !($this->getTocando())) {
            $this->setTocando(true);
        }
    }
    public function pause() {
        if ($this->getLigado() && ($this->getTocando())) {
            $this->setTocando(false);
        }
    }
}